src/views/Login/Ticket.js

```diff
@@ -1,187 +1,209 @@
   const { setPointCursor } = usePointCursor();
   const impliedPoint = useImpliedPoint();
 
-  // point
-  const [pointInput, { data: pointName }] = usePointInput({
-    name: 'point',
-    initialValue: impliedPoint || '',
-    autoFocus: true,
-  });
-
-  // passphrase
-  const [passphraseInput, { data: passphrase }] = usePassphraseInput({
-    name: 'passphrase',
-    label: 'Wallet Passphrase',
-  });
-
-  const [hasPassphraseInput, { data: hasPassphrase }] = useCheckboxInput({
-    name: 'has-passphrase',
-    label: 'Passphrase',
-    initialValue: false,
-  });
-
-  // ticket
-  const [error, setError] = useState();
-  const [deriving, setDeriving] = useState(false);
-  const [ticketInput, { data: ticket, pass: validTicket }] = useTicketInput({
-    name: 'ticket',
-    label: 'Master Ticket',
-    error,
-    deriving,
-  });
-
-  // shards
-  const [shardsInput, { data: isUsingShards }] = useCheckboxInput({
-    name: 'shards',
-    label: 'Shards',
-    initialValue: false,
-  });
-
-  const [shard1Input, { data: shard1, pass: shard1Pass }] = useTicketInput({
-    name: 'shard1',
-    label: 'Shard 1',
-  });
-
-  const [shard2Input, { data: shard2, pass: shard2Pass }] = useTicketInput({
-    name: 'shard2',
-    label: 'Shard 2',
-  });
-
-  const [shard3Input, { data: shard3, pass: shard3Pass }] = useTicketInput({
-    name: 'shard3',
-    label: 'Shard 3',
-  });
-
-  const shardsReady = shard1Pass && shard2Pass && shard3Pass;
-
-  // TODO: maybe want to do this only on-go, because wallet derivation is slow...
-  const deriveWalletFromTicket = useCallback(async () => {
-    // clear states
-    setError();
-    setDeriving(true);
-    setUrbitWallet(Nothing());
-
-    if (
-      !ticket ||
-      !pointName ||
-      !ob.isValidPatq(ticket) ||
-      !ob.isValidPatp(pointName)
-    ) {
-      setDeriving(false);
-      return;
-    }
-
-    const _contracts = need.contracts(contracts);
-    const point = patp2dec(pointName);
-    const urbitWallet = await urbitWalletFromTicket(ticket, point, passphrase);
-    const [isOwner, isTransferProxy] = await Promise.all([
-      azimuth.azimuth.isOwner(
-        _contracts,
-        point,
-        urbitWallet.ownership.keys.address
-      ),
-      azimuth.azimuth.isTransferProxy(
-        _contracts,
-        point,
-        urbitWallet.ownership.keys.address
+  const cachedUrbitWallet = useRef(Nothing());
+
+  const validateForm = useCallback(
+    async (values, errors) => {
+      if (errors.point) {
+        return errors;
+      }
+
+      if (values.useShards) {
+        if (errors.shard1 || errors.shard2 || errors.shard3) {
+          return errors;
+        }
+
+        // shards
+        try {
+          const ticket = kg.combine([
+            values.shard1,
+            values.shard2,
+            values.shard3,
+          ]);
+          const point = patp2dec(values.point);
+          cachedUrbitWallet.current = await urbitWalletFromTicket(
+            ticket,
```

Comment:
I think we can deduplicate this logic with the ticket flow? Looks like the ticket flow does `isOwner, isTransferProxy` checks also, which this doesn't do for some reason.